core/src/main/java/org/mule/runtime/core/internal/metadata/cache/DefaultPersistentMetadataCacheManager.java

```diff
@@ -44,23 +48,52 @@
  *
  * @since 4.1.4, 4.2.0
  */
-public class DefaultPersistentMetadataCacheManager implements MetadataCacheManager, Startable {
+public class DefaultPersistentMetadataCacheManager implements MetadataCacheManager, Initialisable, Startable {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(DefaultPersistentMetadataCacheManager.class);
   public static final String PERSISTENT_METADATA_SERVICE_CACHE = "_mulePersistentMetadataService";
   public static final String MULE_METADATA_CACHE_ENTRY_TTL = SYSTEM_PROPERTY_PREFIX + "metadata.cache.entryTtl.minutes";
   public static final String MULE_METADATA_CACHE_EXPIRATION_INTERVAL =
       SYSTEM_PROPERTY_PREFIX + "metadata.cache.expirationInterval.millis";
 
+  /**
+   * Default implementation should use an {@link ObjectStoreManager} that is tied to the deployable artifact lifecyle.
+   */
   @Inject
   @Named(OBJECT_STORE_MANAGER)
   private ObjectStoreManager objectStoreManager;
 
+  /**
+   * Default implementation should a {@link LockFactory} that comes from the deployavle artifact context.
+   */
   @Inject
```

Comment:
shouldn't we name all injected parameters to avoid errors if more than one is in the registry?

Reply:
Yes, but it is not named AFAIK the lock factory in registry right now.